test/version_test.exs

```diff
@@ -33,86 +33,118 @@ defmodule GitOps.Test.VersionTest do
+    assert new_version("0.1.1", [minor()], pre_release: "alpha") == "0.2.0-alpha"
   end
 
   test "if a pre_release is specified, you get the next version tagged with that pre-release with a major change" do
-    assert new_version(["0.1.1"], [break()], pre_release: "alpha") == "1.0.0-alpha"
+    assert new_version("0.1.1", [break()], pre_release: "alpha") == "1.0.0-alpha"
   end
 
   test "if a pre_release is performed after a pre_release, and the version would not change then it is unchanged" do
-    assert new_version(["0.1.1", "0.1.2-alpha"], [patch()], pre_release: "beta") == "0.1.2-beta"
+    assert new_version("0.1.2-alpha", [patch()], pre_release: "beta") == "0.1.2-beta"
   end
 
   test "if a pre_release is performed after a pre_release, and the version would change, then it is changed" do
-    assert new_version(["0.1.1", "0.1.2-alpha"], [minor()], pre_release: "beta") == "0.2.0-beta"
+    assert new_version("0.1.2-alpha", [minor()], pre_release: "beta") == "0.2.0-beta"
+  end
+
+  test "if a pre_release is performed after using rc, and the version would change, then it is changed with pre-release" do
+    assert new_version("0.1.2-rc0", [patch()], pre_release: "alpha") == "0.1.3-alpha"
+  end
+
+  test "a release candidate starts at 0 if requested on patch" do
+    assert new_version("0.1.0", [patch()], rc: true) == "0.1.1-rc0"
+  end
+
+  test "a release candidate starts at 0 if requested on minor" do
+    assert new_version("0.1.0", [minor()], rc: true) == "0.2.0-rc0"
+  end
+
+  test "a release candidate starts at 0 if requested on break" do
+    assert new_version("0.1.0", [break()], rc: true) == "1.0.0-rc0"
   end
 
-  test "a release candidate starts at 0 if requested" do
-    assert new_version(["0.1.0"], [patch()], rc: true) == "0.1.1-rc0"
+  test "a release candidate increments by one on patch" do
+    assert new_version("0.1.1-rc0", [patch()], rc: true) == "0.1.1-rc1"
   end
 
-  test "a release candidate increments by one as long as the version would normally change" do
-    assert new_version(["0.1.0", "0.1.1-rc0"], [patch()], rc: true) == "0.1.1-rc1"
+  test "a release candidate resets on minor" do
+    assert new_version("0.1.1-rc0", [minor()], rc: true) == "0.2.0-rc0"
   end
 
-  test "a release candidate resets if the version would change more than it originally did" do
-    assert new_version(["0.1.0", "0.1.1-rc0"], [break()], rc: true) == "1.0.0-rc0"
+  test "a release candidate resets on major" do
+    assert new_version("0.1.1-rc0", [break()], rc: true) == "1.0.0-rc0"
   end
 
   test "a release candidate raises correctly when it would not change" do
     assert_raise RuntimeError, ~r/No changes should result in a new release version./, fn ->
-      new_version(["0.1.0", "0.1.1-rc0"], [chore()], rc: true)
+      new_version("0.1.1-rc0", [chore()], rc: true)
     end
   end
 
   test "if a prefix is configured, it is ignored when searching for a tag" do
```

Comment:
Since we aren't searching for a tag, this can go away.

Reply:
Yup, thought so